app/controllers/api/v2/application_controller.rb

```diff
@@ -13,6 +15,10 @@ def authenticate
         end
       end
 
+      def jsonapi_404
+        render jsonapi: nil, status: 404
+      end
```

Comment:
I'm not a fan of this.

This works so long as what's failing is finding the resource specified in the URL, but as soon as a piece of logic further along fails to find something, we'll be at risk of getting 404 errors with nothing being logged to Sentry and no way to debug this without replicating this locally.

I think that instead of doing this, our `build_*` methods in the controllers should catch a `RecordNotFound` error and return the error. e.g.:

```
def build_access_request
  @access_request = authorize AccessRequest.requested.find(params[:id])
rescue ActiveRecord::RecordNotFound => ex
  render jsonapi_errors: ex.message, status: :not_found
end
```
NB. the `jsonapi_errors` bits are a guess, not 100% of the correct way to form that.

Reply:
I've never had an issue with this or seen this become a problem. I've never seen a project that doesn't handle 404s in this way actually. But I don't really care either way.

Reply:
titbits

'out of the box experience'
show_detailed_exceptions

`ex.message be careful not to leak something that should not be leaked`
render `jsonapi_errors: ex.message`, status: :not_found

I have no ideas...

but had experience a number of handling exceptions (including the leaky bit as well)

Reply:
@misaka so the ask here is to move the rescue out of the application controller and into the individual controller actions, case by case?

Reply:
The goal of this PR was just getting 404s to work, they can have an empty body. We should do another PR to introduce error objects as per the JSONAPI spec.

I'm think `render_errors` is meant for validation errors by the way, needs investigating.